src/gc/impl/conservative/gc.d

```diff
@@ -2557,17 +2585,124 @@ struct Gcx
         return recoverPool[bin] = poolIndex < npools ? cast(SmallObjectPool*)pool : null;
     }
 
+    version (COLLECT_FORK)
+    void disableFork() nothrow
+    {
+        markProcPid = 0;
+        shouldFork = false;
+    }
+
+    version (COLLECT_FORK)
+    ChildStatus collectFork(bool block) nothrow
+    {
+        typeof(return) rc = wait_pid(markProcPid, block);
+        final switch (rc)
+        {
+            case ChildStatus.done:
+                debug(COLLECT_PRINTF) printf("\t\tmark proc DONE (block=%d)\n",
+                                                cast(int) block);
+                markProcPid = 0;
+                // process GC marks then sweep
+                thread_suspendAll();
+                thread_processGCMarks(&isMarked);
+                thread_resumeAll();
+                break;
+            case ChildStatus.running:
+                debug(COLLECT_PRINTF) printf("\t\tmark proc RUNNING\n");
+                if (!block)
+                    break;
+                // Something went wrong, if block is true, wait() should never
+                // return RUNNING.
+                goto case ChildStatus.error;
+            case ChildStatus.error:
+                debug(COLLECT_PRINTF) printf("\t\tmark proc ERROR\n");
+                // Try to keep going without forking
+                // and do the marking in this thread
+                break;
+        }
+        return rc;
+    }
+
+    version (COLLECT_FORK)
+    ChildStatus markFork(bool nostack, bool block, bool doParallel) nothrow
+    {
+        // Forking is enabled, so we fork() and start a new concurrent mark phase
+        // in the child. If the collection should not block, the parent process
+        // tells the caller no memory could be recycled immediately (if this collection
+        // was triggered by an allocation, the caller should allocate more memory
+        // to fulfill the request).
+        // If the collection should block, the parent will wait for the mark phase
+        // to finish before returning control to the mutator,
+        // but other threads are restarted and may run in parallel with the mark phase
+        // (unless they allocate or use the GC themselves, in which case
+        // the global GC lock will stop them).
+        // fork now and sweep later
+        import core.stdc.stdio : fflush;
+        import core.stdc.stdlib : _Exit;
+        fflush(null); // avoid duplicated FILE* output
```

Comment:
I wonder if `fflush` is necessary here at all as writing a number of file buffers to disk can be an expensive operation. According to the documentation of `_Exit` no cleanup is done, and as no FILE i/o is happening in the child process there should be no duplications. The only exception are debug `printf`s, so maybe restrict calling `fflush` with the respective conditions or just on `stdout`?